README.md

```diff
@@ -220,87 +220,99 @@ Cloudant({url:"companycloudant.local", username:"somebody", password:"somebody's
-back to your app and can be used to stream data using the Node.js [Stream API](https://nodejs.org/api/stream.html).
-2. `promises` - if you'd prefer to write code in the Promises style then the "promises" plugin turns each request into a Promise. This plugin cannot be used to
-stream data because instead of returning the HTTP request, we are simply returning a Promise instead.
-3. `retry` - on occasion, Cloudant's multi-tenant offerring may reply with an HTTP 429 response because you've exceed the number of API requests in a given amount of time.
-The "retry" plugin will automatically retry your request with exponential back-off. The 'retry' plugin can be used to stream data.
-4. `cookieauth` - this plugin will automatically swap your Cloudant credentials for a cookie transparently for you. It will handle the authentication for you
-and ensure that the cookie is refreshed. The 'cookieauth' plugin can be used to stream data.
-5. custom plugin - you may also supply your own function which will be called to make API calls.
+1. `cookieauth`
 
-#### The 'promises' Plugins
+   This plugin will automatically swap your Cloudant credentials for a cookie. It will handle the authentication for you and ensure that the cookie is refreshed when required.
+2. `promises`
 
-When initialising the Cloudant library, you can opt to use the 'promises' plugin:
+   If you'd prefer to write code in the _Promises_ style then the `promises` plugin turns each request into a _Promise_.
+3. `retry429` (or `retry`)
+
+   On occasion Cloudant's multi-tenant offering may reply with an HTTP 429 response because you've exceed the number of API requests in a given amount of time. The "retry429" plugin will automatically retry your request with an exponential back-off.
+4. `retry5xx`
+
+   The plugin will automatically retry 5xx responses with an exponential back-off.
+5. `retryerror`
+
+   The plugin will automatically retry request errors (e.g. connection reset errors) with an exponential back-off.
+
+_Want to use multiple plugins?_ Not a problem. Simply pass the plugins as an array instead:
 
 ```js
-var cloudant = Cloudant({url: myurl, plugin:'promises'});
+var cloudant = new Cloudant({ url: myurl, plugins: [ 'cookieauth', 'promises', 'retry5xx' ] });
 var mydb = cloudant.db.use('mydb');
+mydb.get('mydoc', function(err, data) {
+  console.log(data.toString('utf8');
+});
 ```
 
-Then the library will return a Promise for every asynchronous call:
+The plugins are _always_ executed in the order they are specified. Remember that all plugins are respected. If one requests a retry then it cannot be overruled by another. If two plugins request different delay times before the next retry attempt then the largest delay time is honoured.
+
+Be aware that if you don't specify any plugins then the `cookieauth` plugin will automatically be added. To disable all plugins you can pass an empty array as the plugin list, i.e. `Cloudant({ url: myurl, plugins: [] })`.
```

Comment:
Is the cookie auth plugin added if no username/password is set?

Reply:
y - it can still be added but doesn't do anything. will update docs accordingly.